utils/unittest/fixtures.go

```diff
@@ -466,6 +469,47 @@ func HeaderBodyFixture(opts ...func(header flow.HeaderBody)) flow.HeaderBody {
 	return header
 }
 
+// WithRootDefaults zeroes out all parent‐QC fields and enforces root constraints.
+func WithRootDefaults() func(*flow.UntrustedHeaderBody) {
+	ts := time.Unix(1_600_000_000, 0)
+	return func(u *flow.UntrustedHeaderBody) {
+		u.ChainID = flow.Emulator // still must be non‐empty
+		u.ParentID = flow.ZeroID  // allowed to be zero
+		u.Height = 0
+		u.View = 0
+		u.Timestamp = ts // non‐zero
+		u.ParentView = 0
+		u.ParentVoterIndices = []byte{}
+		u.ParentVoterSigData = []byte{}
+		u.ProposerID = flow.ZeroID
+		u.LastViewTC = nil
+	}
+}
+
+// UntrustedHeaderBodyFixture returns an UntrustedHeaderBody
+// pre‐populated with sane defaults.  Any opts override those defaults.
+func UntrustedHeaderBodyFixture(opts ...func(*flow.UntrustedHeaderBody)) flow.UntrustedHeaderBody {
```

Comment:
I think it's adding a function which is not needed a closely mimics `HeaderBodyFixture`. Since it's essentially the same type I advice against introducing a new function in widely usage package for this. Instead, I would use `HeaderBodyFixture` and cast value to `UntrustedHeaderBody`, alternatively you can create a helper function in the test package where you are using it that will do the cast, but adding this to a general package is an overkill in my opinion.